src/main/java/org/spongepowered/api/network/packet/Packet.java

```diff
@@ -22,28 +22,29 @@
  * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
  * THE SOFTWARE.
  */
-package org.spongepowered.api.network;
+package org.spongepowered.api.network.packet;
+
+import org.spongepowered.api.network.ChannelBuf;
 
 /**
- * A message transmitted over the connection of a client and a server.
+ * A packet transmitted over the connection of a client and a server.
  *
- * <p>Note to plugin implementations: This must have a publicly accessible
- * no-args constructor.</p>
+ * <p>Note to plugin implementations: This must have a no-args constructor.</p>
```

Comment:
Why remove `publicly accessible` here?

Reply:
Why empty constructor? Maybe inject it via guice?

Reply:
There must be a empty constructor, so the fields can be read from a `ChannelBuf.` And I removed the requirement to make it public, because you can instantiate classes with private constructors without any (or minimal) performance penalty (e.g. with https://github.com/LanternPowered/Lmbda/).

Reply:
Then maybe separate packet serializer and packet data?

Reply:
Possibly, I could provide a `Codec` like in lantern, but you will end up with a lot more classes. I just kept it like this simplicity, and because it was like this before.
@Zidane Thoughts?

Reply:
I want to keep it together, I don't see us needing to do Codec.

If we wrote the server, I'd reconsider.